pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -387,6 +387,10 @@ func BuildHTTPRoutesForVirtualService(
 			}
 		}
 		if catchall {
+			// log the skipped remaining routes in vs.Http to avoid silent skip
+			if i < len(vs.Http)-1 {
+				log.Warnf("Below routes are skipped due to the catch all route : %v", vs.Http[i+1:])
```

Comment:
I think this %v will print a super huge string. `Http` is a big structure. Leave this to other developers.

Reply:
It won't print if there is no http match gets bypassed, if it prints, then the huge string is bypassed, it will help for the troubleshooting since there was a breaking change made.

Reply:
I don't think we want to print this big message. I am not sure we want to print anything at all; istiod logs are not intended for app developer persona. Maybe this is better suited as an analyzer or warning in the webhook

Reply:
I donot mind if you change to debug, and , maybe viable to add the check  in validation webhook

Reply:
The background is that it took us some time to figure out why some VirtualService [routes](https://github.com/istio/istio/issues/46410#issue-1842913565) were working in Isito 1.10 but got 404 after upgrading to `1.17`.  There isn't any log in istiod indicates the problem and I'd believe others will encounter this issue during the upgrade.

I changed to print the skipped route number instead of route itself, PTAL.

Change to debug might not help much since normally we won't run istio in debug mode and validation webhook seems feasible but I think we still can only log a warming message for backward compatible. 
cc @howardjohn  @hzxuzhonghu

Reply:
I can get your meaning, can you update validation as well

Reply:
Added validation, PTAL